### PR TITLE
fix issues with IPv6 canary

### PR DESCRIPTION
**Pull request by abhipth**
**What type of PR is this?**
This addresses bunch of failures encountered during IPv6 Canary tests.

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/lib/add-on.sh [hunk 1]
```diff
@@ -69,7 +82,13 @@ function install_add_on() {
...
 
   echo "installing addon $new_addon_version"
-  aws eks create-addon $ENDPOINT_FLAG --cluster-name "$CLUSTER_NAME" --addon-name $VPC_CNI_ADDON_NAME --resolve-conflicts OVERWRITE --addon-version $new_addon_version --region "$REGION"
+
+  # If installing the addon after deleting it once, we should retain the Service Account Role ARN of the Addon
+  if [ "$EXISTING_SERVICE_ACCOUNT_ROLE_ARN" != "null" ]; then
+     SA_ROLE_ARN_ARG="--service-account-role-arn $EXISTING_SERVICE_ACCOUNT_ROLE_ARN"
+  fi
+
+  aws eks create-addon $ENDPOINT_FLAG $SA_ROLE_ARN_ARG --cluster-name "$CLUSTER_NAME" --addon-name $VPC_CNI_ADDON_NAME --resolve-conflicts OVERWRITE --addon-version $new_addon_version --region "$REGION"
```
Comment (jayanthvn) [inline scripts/lib/add-on.sh:91]: if `$EXISTING_SERVICE_ACCOUNT_ROLE_ARN` is null then for the create-addon we would pass `$SA_ROLE_ARN_ARG` null, is that fine?

Comment (abhipth) [reply scripts/lib/add-on.sh:91]: I think $SA_ROLE_ARN_ARG would be an empty string which I believe should be fine, but I have not tested it :D

Comment (jayanthvn) [reply scripts/lib/add-on.sh:91]: right, it would be empty. Should be fine then.

**Review by jayanthvn — APPROVED** [review]
lgtm

File: scripts/lib/add-on.sh (hunk 1 above)
Comment (achevuru) [reply scripts/lib/add-on.sh:91]: Yeah should count as an extra space

**Pull request merged**